time_stamper_cv/src/image_processor.cpp

```diff
@@ -25,20 +27,24 @@ cv_bridge::CvImage ImageProcessor::process(const sensor_msgs::Image &image) {
   convex_shape_->process(points);
   convex_shape_->pollShapeStatus(ImageProcessor::log);
 
-  int number = -1;
   if (convex_shape_->isShapeValid()) {
     led_parser_->processImage(input_mat);
 
-    /*
-     * Get inverted homography, so we know the position of each LED even if it's turned off.
-     */
-    led_parser_->transformLedRow(convex_shape_->getInvHomography());
+    //Get inverted homography, so we know the position of each LED even if it's turned off.
+    led_parser_->transformLedRow("BottomRow", convex_shape_->getInvHomography());
+    led_parser_->transformLedRow("TopRow", convex_shape_->getInvHomography());
 
-    number = led_parser_->getBinaryValue();
+    /* When the first and last led in the top row are on, the state of the bottom row is changing.
+       The frame has to be skipped. */
+    if (led_parser_->isLedOn("TopRow", 0) && led_parser_->isLedOn("TopRow", 9)) {
```

Comment:
The `9` seems to be derived from the hardware layout. I'd recommend deriving such numbers from some hardware constants such as "LEDs in a row", so it is easier to adapt the code for different hardware revisions and help the reader to understand where it comes from. As a general rule, try to avoid unexplained numbers in the code.